### PR TITLE
Makes unit_association work with new and old style unit definitions.

### PR DESCRIPTION
**Pull request by bmbouter**

### CODE REVIEW

File: server/pulp/server/db/model/__init__.py [hunk 1]
```diff
@@ -482,6 +482,33 @@ def unit_key_as_named_tuple(self):
...
+    @property
+    def type_id(self):
+        """
+        Backwards compatible interface for unit_type_id
+
+        The pre-mongoengine units used type_id to track what is stored in unit_type_id. This
+        provides internal backwards compatibility allowing code to not be updated until all models
+        are converted to mongoengine and able to use the new name exclusively.
+
+        This should be remove
```
Comment (jortel) [inline server/pulp/server/db/model/__init__.py]: s/remove/removed

File: server/pulp/server/managers/repo/unit_association.py [hunk 2]
```diff
@@ -183,32 +185,45 @@ def associate_from_repo(source_repo_id, dest_repo_id, criteria=None,
...
+        source_repo_unit_types = set(source_repo.content_unit_counts.keys())
 
         # Now we can make sure the destination repository's importer is capable
         # of importing either the selected units or all of the units
-        associated_unit_type_ids = calculate_associated_type_ids(source_repo_id, associate_us)
-        unsupported_types = [t for t in associated_unit_type_ids if t not in supported_type_ids]
-
-        if len(unsupported_types) > 0:
-            raise exceptions.InvalidValue(['types'])
+        if not source_repo_unit_types.issubset(supported_type_ids):
```
Comment (jortel) [inline server/pulp/server/managers/repo/unit_association.py]: Nice usage of sets :)

**Comment by jortel** [comment]
Changes look good overall but test coverage went down with this PR.  Although, 
```
(master)
pulp.server.managers.repo.unit_association                                    156      5    97%   327, 431-434
```
To:
```
#2019
pulp.server.managers.repo.unit_association                                    162     23    86%   197, 205-226, 342, 398-404, 411-412, 446-449
```

**Comment by jortel** [comment]
It would be good if the commit contained 'ref #2019', right?

**Comment by bmbouter** [comment]
I spent some time trying to fix the removed test but it is not really a unit test and is difficult to fix. Once we get the functional aspects of the mongoengine conversion done we should revisit the test coverage.

File: server/pulp/server/db/model/__init__.py (hunk 1 above)
Comment (bmbouter) [inline server/pulp/server/db/model/__init__.py]: This line wasn't even finished. I finished it as:

```
This should be removed once the old, non-mongoengine code paths are removed.
```

**Comment by jortel** [comment]
Fair enough.  Merge on.

**Comment by bmbouter** [comment]
@jortel thanks for the review

**Pull request merged**